In [3]:
# Task 2.1
def Converter(DenaryNumber):
    if DenaryNumber == 0 or DenaryNumber == 1:
        print(DenaryNumber, end='')
    else:
        print(DenaryNumber % 2, end='')
        Converter(DenaryNumber // 2)

In [4]:
Converter(56)

000111

# Task 2.2
---
The output binary sequence is inverted (i.e. the number 56 should be `111000` not `000111`). The error is that we need to flip the output before displaying it.

In [7]:
# Task 2.3
def Converter(DenaryNumber, ret=''):
    if DenaryNumber == 0 or DenaryNumber == 1:
        ret = str(DenaryNumber) + ret
        print(ret)
    else:
        ret = str(DenaryNumber % 2) + ret
        Converter(DenaryNumber // 2, ret)

In [8]:
Converter(56)

111000


# Task 2.3
|DenaryNumber|Purpose of the test|Expected output|
|---|---|---|
|56|Normal Case|111000|
|1|Boundary Case|1|
|AAA|Invalid Case|Error|

In [11]:
Converter(56)

111000


In [10]:
Converter(1)

1
